it/controllers/benefits/reimbursed/NonTaxableCostsBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,540 @@
+          radioButtonCheck(yesText, 1, Some(true))
+          radioButtonCheck(noText, 2, Some(false))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render 'non taxable costs' page with the correct content with no pre-filled" which {
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+              benefits = Some(benefits(fullReimbursedCostsVouchersAndNonCashModel.copy(expensesQuestion = Some(false)))))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlGet(pageUrl(taxYearEOY), welsh = user.isWelsh,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          import Selectors._
+          import user.commonExpectedResults._
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          textOnPageCheck(expectedCaption, captionSelector)
+          textOnPageCheck(user.specificExpectedResults.get.paragraph, paragraphSelector)
+          radioButtonCheck(yesText, 1, Some(false))
+          radioButtonCheck(noText, 2, Some(true))
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+      }
+    }
+
+    "redirect to another page when the request is valid but they aren't allowed to view the page and" should {
+
+      val user = UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN))
+
+      "redirect the user to the check employment benefits page when the benefitsReceived question is false" which {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          authoriseAgentOrIndividual(isAgent = false)
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+            benefits = Some(BenefitsViewModel(isUsingCustomerData = true)))), userRequest)
+          urlGet(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe
+            Some(s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/check-employment-benefits?employmentId=$employmentId")
```

Comment:
This can be changed to use the reverse routes:

Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url

The  same for the  other occurrences within the ISPEC too